python/pylibcugraph/pylibcugraph/uniform_neighborhood_sampling.pyx

```diff
@@ -0,0 +1,190 @@
+                               start_info_list,
+                               h_fan_out,
+                               bool_t with_replacement,
+                               bool_t do_expensive_check):
+    """
+    Does uniform neighborhood sampling.
+
+    Parameters
+    ----------
+    resource_handle: ResourceHandle
+        Handle to the underlying device and host resources needed for
+        referencing data and running algorithms.
+
+    input_graph: MGGraph
+        The input graph, for Multi-GPU operations.
+
+    start_info_list: device array type
+        Device array containing the list of starting vertices for sampling,
+        as well as starting labels for reorganizing the results after the
+        sending the input to different callers
+
+    fanout_vals: list
+        Host array containing the fan out at each step in the sampling
+        algorithm
+
+    with_replacement: bool
+        If true, sampling procedure is done with replacement (the same vertex
+        can be selected multiple times in the same step)
+
+    do_expensive_check: bool
+        If True, performs more extensive tests on the inputs to ensure
+        validitity, at the expense of increased run time.
+
+    Returns
+    -------
+    A tuple of device arrays, where the first and second items in the tuple
+    are device arrays containing the starting and ending vertices of each
+    walk respectively, the third item in the tuple is a device array
+    containing the start labels, the fourth item in the tuple is a device
+    array containing the indices for ..., and the fifth item in the tuple
+    is a device containing the counts for ...
+
+    Examples
+    --------
+    >>> import pylibcugraph, cupy, numpy
+    >>> srcs = cupy.asarray([0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4],
+    ...                     dtype=numpy.int32)
+    >>> dsts = cupy.asarray([1, 2, 3, 4, 5, 6, 7, 5, 6, 7, 5, 6, 7],
+    ...                     dtype=numpy.int32)
+    >>> weights = cupy.asarray([1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,
+    ...                         1.0, 1.0, 1.0, 1.0], dtype=numpy.float32)
+    >>> resource_handle = pylibcugraph.experimental.ResourceHandle()
+    >>> graph_props = pylibcugraph.experimental.GraphProperties(
+    ...     is_symmetric=False, is_multigraph=False)
+    >>> G = pylibcugraph.experimental.MGGraph(
+    ...     resource_handle, graph_props)
+
+    """
+    print("Hello from uniform_neighborhood_sampling.pyx!")
+    
+    cdef cugraph_resource_handle_t* c_resource_handle_ptr = \
+        resource_handle.c_resource_handle_ptr
+    cdef cugraph_graph_t* c_graph_ptr = input_graph.c_graph_ptr
+
+    cdef cugraph_sample_result_t* result_ptr
+    cdef cugraph_error_code_t error_code
+    cdef cugraph_error_t* error_ptr
+
+    cdef uintptr_t cai_start_ptr = \
+        start_info_list.__cuda_array_interface__["data"][0]
+    cdef uintptr_t cai_labels_ptr = \
+        start_info_list.__cuda_array_interface__["data"][1]
```

Comment:
If you combined these two variables into one `start_info_list`, this won't work because `__cuda_array_interface__["data"][1]` refers to something totally different I think. Based on this[ documentation](https://numba.pydata.org/numba-doc/dev/cuda/cuda_array_interface.html), `__cuda_array_interface__["data"][1]` is read-only flag as a Python bool

Reply:
This has been separated into two variables in pylibcugraph. It is still combined as `start_info_list`in the cugraph layer